state/factory/workingset.go

```diff
@@ -169,7 +170,11 @@ func (ws *workingSet) runAction(
 	// for replay tx, check against deployer whitelist
 	g := genesis.MustExtractGenesisContext(ctx)
 	if selp.Encoding() == uint32(iotextypes.Encoding_ETHEREUM_UNPROTECTED) && !g.IsDeployerWhitelisted(selp.SenderAddress()) {
-		return nil, errors.Errorf("replay deployer %v not whitelisted", selp.SenderAddress().String())
+		if protocol.MustGetFeatureCtx(ctx).PurgeActpoolOnRunActionError {
+			return nil, errors.Wrap(errDeployerNotWhitelisted, selp.SenderAddress().String())
+		} else {
+			return nil, errors.Errorf("replay deployer %v not whitelisted", selp.SenderAddress().String())
```

Comment:
it is not a hard fork. If an error is returned during block mint, the action will not be included. If an error is returned during block validation, the block will be rejected

Reply:
yeah, fixed